Mathlib/Order/InitialSeg.lean

```diff
@@ -29,31 +33,21 @@ These notations belong to the `InitialSeg` locale.
 * `r ≺i s`: the type of principal segment embeddings of `r` into `s`.
 -/
 
-
-/-!
-### Initial segments
-
-Order embeddings whose range is an initial segment of `s` (i.e., if `b` belongs to the range, then
-any `b' < b` also belongs to the range). The type of these embeddings from `r` to `s` is called
-`InitialSeg r s`, and denoted by `r ≼i s`.
--/
+/-! ### Initial segment embeddings -/
 
 variable {α : Type*} {β : Type*} {γ : Type*} {r : α → α → Prop} {s : β → β → Prop}
   {t : γ → γ → Prop}
 
 open Function
 
 /-- If `r` is a relation on `α` and `s` in a relation on `β`, then `f : r ≼i s` is an order
-embedding whose range is an initial segment. That is, whenever `b < f a` in `β` then `b` is in the
+embedding whose `Set.range` is a lower set. That is, whenever `b < f a` in `β` then `b` is in the
 range of `f`. -/
 structure InitialSeg {α β : Type*} (r : α → α → Prop) (s : β → β → Prop) extends r ↪r s where
-  /-- The order embedding is an initial segment -/
-  init' : ∀ a b, s b (toRelEmbedding a) → ∃ a', toRelEmbedding a' = b
+  /-- The range of the embedding is a lower set. -/
```

Comment:
We write `lower set` here, but this doesn't use the predicate [IsLowerSet](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/UpperLower/Basic.html#IsLowerSet). We should either
a) use the existing definition
b) prove the existing definition is equivalent to this and reference that proof in this docstring
c) give an alternative informal description

The choice is yours :)

Reply:
We can't use `IsLowerSet` since this is for an arbitrary relation, and `IsLowerSet` requires `≤`. But morally, that's what it should be.

Reply:
Good point - that rules out a) but giving the special case for orders would work for b), and c) is my preferred one anyway

Reply:
Working with initial and principal segments with `<` is somewhat annoying, and it often requires writing the types explicitly like `((· < ·) : α → α → Prop) ≼i ((· < ·) : β → β → Prop)`. I think that issue should probably be fixed before we introduce any API for this special case.

Do I just copy the description from the structure docstring? The "whenever `b < f a` in `β` then..." bit.

Reply:
Actually, I will introduce the `IsLowerSet` version of this in a separate PR. That way we can use "proper" vocabulary and have it be right!

Reply:
Two months later, and we now have [`InitialSeg.isLowerSet_range`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/InitialSeg.html#InitialSeg.isLowerSet_range).